pkg/depman/links.go

```diff
@@ -0,0 +1,65 @@
+package depman
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/acronis/go-cti/pkg/bundle"
+)
+
+const (
+	RAMLExt = ".raml"
+)
+
+var patchDepsRe = regexp.MustCompile(fmt.Sprintf(`(?:..\/)*(%s)\/`, strings.ReplaceAll(DependencyDirName, ".", `\.`)))
+var patchRamlxRe = regexp.MustCompile(fmt.Sprintf(`(?:..\/)*(%s)\/`, strings.ReplaceAll(bundle.RamlxDirName, ".", `\.`)))
+
+func replaceCaptureGroup(reg *regexp.Regexp, input, replacement string, groupIndex int) string {
+	matches := reg.FindAllStringSubmatchIndex(input, -1)
+	if matches == nil {
+		return input
+	}
+
+	sb := strings.Builder{}
+	prevEnd := 0
+	for _, match := range matches {
+		sb.WriteString(input[prevEnd:match[2*groupIndex]])
+		sb.WriteString(replacement)
+		prevEnd = match[2*groupIndex+1]
+	}
+	sb.WriteString(input[prevEnd:])
+
+	return sb.String()
+}
+
+func patchRelativeLinks(dir string) error {
+	if err := filepath.WalkDir(dir, func(file string, dir fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if filepath.Ext(dir.Name()) != RAMLExt {
+			return nil
+		}
+
+		raw, err := os.ReadFile(file)
+		if err != nil {
+			return fmt.Errorf("read file: %w", err)
+		}
+		// implies that dependencies are correct in the first place
+		content := replaceCaptureGroup(patchDepsRe, string(raw), "../../"+DependencyDirName, 1)
+		content = replaceCaptureGroup(patchRamlxRe, content, "../../"+bundle.RamlxDirName, 1)
```

Comment:
Why not just:
```
patchDepsRe.ReplaceAllLiteralString(raw, "../../$1/")
patchRamlxRe.ReplaceAllString(raw, "../../$1/")
```

Reply:
.dep could be part of other string so regex is more fair